src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseBaseKernels.hpp

```diff
@@ -782,13 +782,6 @@ class ElementBasedAssemblyKernel
 
       }
     }
-
-    // check zero diagonal (works only in debug)
-    for( integer ic = 0; ic < numComp; ++ic )
-    {
-      GEOS_ASSERT_MSG( LvArray::math::abs( stack.localJacobian[ic][ic] ) > minDensForDivision,
-                       GEOS_FMT( "Zero diagonal in Jacobian: equation {}, value = {}", ic, stack.localJacobian[ic][ic] ) );
-    }
```

Comment:
Would we want to check `stack.localJacobian[ic][ic+1]` instead?

Reply:
I am not sure...
When I originally added the check, my main concern was the liner solver, but now I don't know.
Maybe @castelletto1 and @victorapm can clarify what should be checked.

Reply:
Zero diagonals can be an issue for the robustness of ILU0, which is used global relaxation in the level where the densities are eliminated.

Would it be possible to change the order of the equations such that we don't have zero diagonals?

Reply:
By default GEOS already doing that, if I remember correctly - first equation will be replaced by sum of all equations and all other equations are shifted - old first becomes new second and so on.
After that:
```
         p      rho_0      rho_1
eq 0   x0+x1    x0+x1        x1
eq 1     x0       x0          0
```
So no zero diag anymore.